admin/js/profiles.livesession.js

```diff
@@ -18,72 +18,105 @@
  *          Oliver Gutiérrez <ogutierrez@redhat.com>
  */
 
-
 var updater;
 var submit=false;
-var tries=0;
-var MAXTRIES=5;
+var sc;
 
 window.alert = function(message) {
-  console.log('ALERT:' + message);
+  console.log('FC: Alert message:' + message);
 }
 
 /* SPICE HTML5 */
-var sc;
-var retrying = null;
+function spiceClientConnection(host, port) {
 
+  var connecting = null;
+  var conn_timeout = 15000; //ms
+  var noretry = false;
+
+  function set_connection_timeout() {
+    if (!connecting) {
+      connecting = setTimeout(function() {
+        if (sc) sc.stop()
+        $('#spice-screen').html('');
+        connecting = null;
+        noretry = true;
+        console.log('FC: Connection tries timed out');
+        $('#spinner-modal').modal('hide');
+        showMessageDialog('Connection error to virtual machine.', 'Connection error');
+      }, conn_timeout);
+    }
+  }
 
-function spiceClientConnection(host, port) {
+  function spice_connected() {
+    console.log('FC: Connected to virtual machine using SPICE');
+    $('#spinner-modal').modal('hide');
+    if (connecting) {
+      clearTimeout(connecting);
+      connecting = null;
+    }
+  }
 
   function spice_error(err) {
+    console.log("FC: SPICE connection error:", err.message);
+
+    set_connection_timeout()
 
-    console.error(err);
-    // Try to reconnect
-    if (!retrying && sc.state != 'ready') {
-      retrying = setTimeout(function() {
-        console.log('Trying to reconnect');
+    if (err.message == 'Unexpected close while ready' || err.message == 'Connection timed out.' || sc.state != 'ready')  {
+      if (!noretry) {
+        $('#spinner-modal h4').text('Connecting to virtual machine. Please wait...');
+        $('#spinner-modal').modal('show');
         do_connection();
-        retrying = null;
-      }, 200);
+      }
+    } else {
+      $('#spinner-modal').modal('hide');
+      if (connecting) {
+        clearTimeout(connecting);
+        connecting = null;
+      }
+      showMessageDialog('Connection error to virtual machine.', 'Connection error');
     }
-  }
 
-  function agent_connected(sc) {
```

Comment:
Probably moved it to the "end" of callback functions because nowadays it's not being used and my idea was to test to enable it again later because I changed some references that could be the reason of disconnections when we enabled it in the past.